appbuilder/providers/device-app-data-provider.ts

```diff
@@ -0,0 +1,339 @@
+			} else {
+				this.$errors.failWithoutHelp(`Unsupported LiveSync version: ${version}`);
+			}
+
+			this._deviceProjectRootPath = this.getDeviceProjectRootPath(util.format(deviceTmpDirFormat, this.appIdentifier));
+		}
+
+		return this._deviceProjectRootPath;
+	}
+
+	public get liveSyncFormat(): string {
+		return null;
+	}
+
+	public encodeLiveSyncHostUri(hostUri: string): string {
+		return hostUri;
+	}
+
+	public getLiveSyncNotSupportedError(): string {
+		return `You can't LiveSync on device with id ${this.device.deviceInfo.identifier}! Deploy the app with LiveSync enabled and wait for the initial start up before LiveSyncing.`;
+	}
+
+	public isLiveSyncSupported(): IFuture<boolean> {
+		return (() => {
+			let isApplicationInstalled = this.device.applicationManager.isApplicationInstalled(this.appIdentifier).wait();
+			if (!isApplicationInstalled) {
+				this.$deployHelper.deploy(this.$devicePlatformsConstants.Android.toLowerCase()).wait();
+			}
+		 	return this.getLiveSyncVersion().wait() !== 0;
+		}).future<boolean>()();
+	}
+
+	private getLiveSyncVersion(): IFuture<number> {
+		return (() => {
+			if (!this._liveSyncVersion) {
+				this._liveSyncVersion = (<Mobile.IAndroidDevice>this.device).adb.sendBroadcastToDevice(CHECK_LIVESYNC_INTENT_NAME, {"app-id": this.appIdentifier}).wait();
+			}
+			return this._liveSyncVersion;
+		}).future<number>()();
+	}
+}
+
+export class AndroidCompanionAppIdentifier extends DeviceAppDataBase implements ILiveSyncDeviceAppData {
+	constructor(_appIdentifier: string,
+		public device: Mobile.IDevice,
+		public platform: string,
+		private $companionAppsService: ICompanionAppsService,
+		private $projectConstants: IProjectConstants) {
+		super($companionAppsService.getCompanionAppIdentifier($projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.Cordova, platform));
+	}
+
+	public get deviceProjectRootPath(): string {
+		return this.getDeviceProjectRootPath(util.format(DEVICE_TMP_DIR_FORMAT_V3, this.appIdentifier));
+	}
+
+	public get liveSyncFormat(): string {
+		return "%s/Mist/MobilePackage/redirect?token=%s";
+	}
+
+	public encodeLiveSyncHostUri(hostUri: string): string {
+		return hostUri;
+	}
+
+	public getLiveSyncNotSupportedError(): string {
+		return `Cannot LiveSync changes to the companion app. The companion app is not installed on ${this.device.deviceInfo.identifier}.`;
+	}
+
+	public isLiveSyncSupported(): IFuture<boolean> {
+		return this.device.applicationManager.isApplicationInstalled(this.appIdentifier);
+	}
+}
+
+export class AndroidNativeScriptCompanionAppIdentifier extends DeviceAppDataBase implements ILiveSyncDeviceAppData {
+	constructor(_appIdentifier: string,
+		public device: Mobile.IDevice,
+		public platform: string,
+		private $companionAppsService: ICompanionAppsService,
+		private $projectConstants: IProjectConstants) {
+		super($companionAppsService.getCompanionAppIdentifier($projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.NativeScript, platform));
```

Comment:
According to the rest of the code, you can just call `super(_appIdentifier)` here